### PR TITLE
Testing

### LINKED ISSUE #3: `recipe-test.js`
- [ ] Line 11: data.recipeData is not a property (needs to be instance of something that has `recipeData` as a property)
- [ ] Line 12: `new Recipe(recipeInfo)`: Recipe should have the recipe data passed into it (recipeInfo) doesn't currently represent that
- [ ] Line 48: there's nothing inside the assertion for the method to be equal to anything)
- [ ] There's no sad path testing
- [ ] It's all in ES5 (not really a deal breaker but still)

### LINKED ISSUE #4: `user-test.js`:
- [x] Line 12: same issue as line 11 in recipe-test
- [x] Line 13: same issue as line 12 in recipe-test
- [x] Line 15: object in recipe variable doesn't reflect the same object properties as the data in recipe-data.js
- [ ] No sad path testing

### PR DESCRIPTION
**Pull request by mistercanderson**
# Description of Changes

- Changed some method/variable/argument names to be more descriptive
- Changed `user-test` to use ES6 syntax
- Fixed bug in `beforeEach` hook that was preventing test from running
- Wrote & tested `calculateIngredientCost` method
- Recipe class now has property that holds ingredient data for each ingredient included in the recipe

# Linked Issues

- closes #4 
- closes #3

### CODE REVIEW

File: test/recipe-test.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ describe('Recipe', function() {
   let recipeInfo;
 
   beforeEach(function() {
-    recipeInfo = data.recipeData[0];
+    recipeInfo = data[0];
```
Comment (rhen92) [inline test/recipe-test.js]: Nice way to access the data!

**Review by rhen92 — APPROVED** [review]
Everything looks good!

File: src/recipe.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
 class Recipe {
-  constructor(recipe) {
```
Comment (battan40) [inline src/recipe.js:2]: We may need to import the ingredient data in the function that calculates that data as a parameter on line 11, lets review again tomorrow when we check the specs

File: src/recipe.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
     this.tags = recipe.tags;
     this.ingredients = recipe.ingredients;
+    this.ingredientData = ingredientData.filter(data => this.ingredients.some(ingredient => ingredient.id === data.id))
   }
+  
   calculateIngredientsCost() {
-    // return this.ingredients.map(i => {
-    //   ingredientData.find(ingredient => ingredient === i);
-    // });
+    const sortedIngredients = this.ingredients.sort((a, b) => a.id - b.id)
```
Comment (battan40) [inline src/recipe.js:12]: This sort method of yours saved the day, honestly, it was a pleasure to work on this one with you. Brilliant.

File: src/recipe.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
+    const sortedIngredientData = this.ingredientData.sort((a, b) => a.id - b.id)
+    return sortedIngredientData.reduce((counter, ingredient, index) => {
```
Comment (battan40) [inline src/recipe.js:14]: Great use of reduces index here.

File: test/recipe-test.js [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
   let recipe;
   let recipeInfo;
+  let ingredientInfo;
 
   beforeEach(function() {
-    recipeInfo = data.recipeData[0];
-    recipe = new Recipe(recipeInfo);
+    recipeInfo = recipeData[0];
+    ingredientInfo = ingredientData
+    recipe = new Recipe(recipeInfo, ingredientInfo);
```
Comment (battan40) [inline test/recipe-test.js:15]: Similar to the corresponding recipe class, We may need to review my idea of changing the class per the specs of the original project.  I think its working perfectly, as it is though.

**Review by battan40 — APPROVED** [review]

**Comment by battan40** [comment]
We might need to revisit a couple of things, and today was a great start.

**Pull request merged**